inject/src/main/java/io/micronaut/inject/ast/utils/EnclosedElementsQuery.java

```diff
@@ -0,0 +1,292 @@
+                    }
+                }
+            }
+            if (result.isOnlyInstance() && element.isStatic()) {
+                return false;
+            } else if (result.isOnlyStatic() && !element.isStatic()) {
+                return false;
+            }
+            if (result.isOnlyAccessible()) {
+                // exclude private members
+                // exclude synthetic members or bridge methods that start with $
+                if (element.isPrivate() || element.getName().startsWith("$")) {
+                    return false;
+                }
+                if (element instanceof MemberElement && !((MemberElement) element).isAccessible()) {
+                    return false;
+                }
+            }
+            if (!result.getModifierPredicates().isEmpty()) {
+                Set<ElementModifier> modifiers = element.getModifiers();
+                for (Predicate<Set<ElementModifier>> modifierPredicate : result.getModifierPredicates()) {
+                    if (!modifierPredicate.test(modifiers)) {
+                        return false;
+                    }
+                }
+            }
+            if (!result.getNamePredicates().isEmpty()) {
+                for (Predicate<String> namePredicate : result.getNamePredicates()) {
+                    if (!namePredicate.test(element.getName())) {
+                        return false;
+                    }
+                }
+            }
+            if (!result.getAnnotationPredicates().isEmpty()) {
+                for (Predicate<AnnotationMetadata> annotationPredicate : result.getAnnotationPredicates()) {
+                    if (!annotationPredicate.test(element)) {
+                        return false;
+                    }
+                }
+            }
+            if (!result.getTypePredicates().isEmpty()) {
+                for (Predicate<ClassElement> typePredicate : result.getTypePredicates()) {
+                    ClassElement ce;
+                    if (element instanceof ConstructorElement) {
+                        ce = classElement;
+                    } else if (element instanceof MethodElement) {
+                        ce = ((MethodElement) element).getGenericReturnType();
+                    } else if (element instanceof ClassElement) {
+                        ce = (ClassElement) element;
+                    } else if (element instanceof FieldElement) {
+                        ce = ((FieldElement) element).getGenericField();
+                    } else {
+                        throw new IllegalStateException("Unknown element: " + element);
+                    }
+                    if (!typePredicate.test(ce)) {
+                        return false;
+                    }
+                }
+            }
+// TODO: FIX only injected
+//                if (result.isOnlyInjected() && !element.hasDeclaredAnnotation(AnnotationUtil.INJECT)) {
+//                    return false;
+//                }
```

Comment:
should this TODO be addressed before merging? Seems like this would be a breaking change otherwise?

Reply:
Actually that’s is not needed anymore, this was applying some extra rules which are now by default

Reply:
perhaps delete the code in that case?

Reply:
Deleted

Reply:
Undeleted it :D found some logic that was added in the query method.